staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/handler.go

```diff
@@ -505,11 +513,25 @@ func (rdm *resourceDiscoveryManager) serveHTTP(resp http.ResponseWriter, req *ht
 	response := cache.cachedResponse
 	etag := cache.cachedResponseETag
 
+	mediaType, _, err := negotiation.NegotiateOutputMediaType(req, rdm.serializer, DiscoveryEndpointRestrictions)
+	if err != nil {
+		// Should never happen. wrapper.go will only proxy requests to this
+		// handler if the media type passes DiscoveryEndpointRestrictions
+		utilruntime.HandleError(err)
+		resp.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	var targetGV schema.GroupVersion
+	if mediaType.Convert != nil {
+		targetGV = mediaType.Convert.GroupVersion()
```

Comment:
We know the only two allowed values.  Can we add a check to be sure we only accept those two values and error in every other case?

Reply:
Added a check. Should also be a never happens case because `wrapper.go` will filter all requests handled by this handler (https://github.com/kubernetes/kubernetes/blob/996d5c2d8c7e44463f26e088d73e179e36f0c7ba/staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/wrapper_test.go)